hybrasyl/Objects/Creature.cs

```diff
@@ -801,6 +815,14 @@ public virtual void RegenerateMp(double mp, Creature regenerator = null)
 
         public virtual void Damage(double damage, Enums.Element element = Enums.Element.None, Enums.DamageType damageType = Enums.DamageType.Direct, Castables.DamageFlags damageFlags = Castables.DamageFlags.None, Creature attacker = null, bool onDeath = true)
         {
+            var attackerUser = attacker as User;
+            if(attackerUser != null)
```

Comment:
This is a little hard to follow, I think the logic can be simplified. All `Creatures` have the `Id` and `Name` property so, I don't think there is a need to check if it can be converted to a user, or to convert them to user; we just need to know it _is_ a user. The last part seems to be enforcing a "tagging" check, which can be done at the main scope. The check  for `ActiveUsersByName` is also incorrect here since I believe you mean to check whether the existing `FirstHitter` is online - not the attacker (presumably the latter is always true. :P)

Similarly I think there is a logic error in the tagging check (we want to make sure the `FirstHitter`'s group contains `attackerUser` - right now it checks if `attackerUser`'s group contains `attackerUser` which, again, will always be true.

To sum up:

```csharp
if ((attacker is User && FirstHitter == null) || (FirstHitter != null && !Game.World.ActiveUsersByName.ContainsKey(FirstHitter.Name))
  _mFirstHitter = attacker.Id;

if (FirstHitter != attacker && (FirstHitter as User).Grouped && (FirstHitter as User).Group.Members.Contains(attacker.Name))
  return;


 
```

Reply:
Yea, I was over thinking this. Essentially, I want the "tagging" to be reset if all the users leave the game or the map to avoid having creatures "tagged" indefinitely. I'll have to think how to do that separately.

Reply:
I think a timeout for this case is more useful rather than trying to keep data around to handle the case indefinitely. I suggest 5 minutes?

Reply:
Okay. Do you want me to create a constant for that timeout (so we can make it a defined field in the config xml or where ever these are going to go?)

Reply:
Logic adjusted.